packages/tools/kadena-cli/src/tx/commands/txLocal.ts

```diff
@@ -0,0 +1,80 @@
+import type { ICommandResult } from '@kadena/client';
+import { createClient } from '@kadena/client';
+import type { ChainId } from '@kadena/types';
+import type { Command } from 'commander';
+import { z } from 'zod';
+import { createCommand } from '../../utils/createCommand.js';
+import { globalOptions } from '../../utils/globalOptions.js';
+import { log } from '../../utils/logger.js';
+import { generateClientUrl } from '../utils/txHelpers.js';
+import { arbitraryCodeTemplate } from './templates/templates.js';
+import { createTransaction } from './txCreateTransaction.js';
+
+export const createTxLocalCommand: (program: Command, version: string) => void =
+  createCommand(
+    'local',
+    'Submit pact code as a local call',
+    [
+      globalOptions.networkOptional({ disableQuestion: true }),
+      globalOptions.chainIdOptional({ disableQuestion: true }),
+    ],
+    async (option, { values }) => {
+      const { network, networkConfig } = await option.network();
+      const { chainId } = await option.chainId();
+      log.debug('tx:local', { network, networkConfig, chainId, values });
+
+      const code = values[0];
+      const templateChainId = (chainId ?? '0') as ChainId;
+      const templateNetworkId = networkConfig?.networkId ?? 'testnet04';
+      const templateNetworkHost =
+        networkConfig?.networkHost ?? 'https://api.testnet.chainweb.com';
+      const templateNetwork = networkConfig?.network ?? 'testnet';
+
+      if (!code) {
+        log.warning(`No code passed to command. use: kadena tx local '(pact)'`);
+        return;
+      }
+
+      const transaction = await createTransaction(
+        arbitraryCodeTemplate(code, templateChainId, templateNetworkId),
+        {},
+      );
+
+      const client = createClient(
+        generateClientUrl({
+          chainId: templateChainId,
+          network: templateNetwork,
+          networkId: templateNetworkId,
+          networkHost: templateNetworkHost,
+          networkExplorerUrl: '',
+        }),
+      );
+
+      const response = await client
+        .local(transaction, {
```

Comment:
I forgot the function but kadena client exposes another function, that has this automatically disabled and does the call , saves u padding 2 options.

Reply:
That's dirtyRead: https://github.com/kadena-community/kadena.js/blob/main/packages/libs/client/src/client/client.ts#L392

Reply:
Personally I like local better, having the option there makes it clear I'm doing a local call with those settings. dirtyRead doesn't mean anything to me. But if it's against Kadena's best practices I can change it...

Reply:
We can go with local dirtyRead in naming doesn’t say anything idd we should actually rename that function to something more obvious in the client. Let’s keep local for now for us